tests/DocBlock/TypeExpressionTest.php

```diff
@@ -1014,6 +1046,26 @@ public static function provideSortTypesCases(): iterable
             '18_446_744_073_709_551_616|-8.2023437675747321e-18_446_744_073_709_551_616',
             '-8.2023437675747321e-18_446_744_073_709_551_616|18_446_744_073_709_551_616',
         ];
+
+        yield 'mixed | and & glue' => [
+            'Foo|Baz&Bar',
+            'Bar&Baz|Foo',
```

Comment:
[Mixed `|` and `&` require parentheses (DNF type)](https://3v4l.org/NNThv), and [is invalid even as a PHPDoc](https://phpstan.org/r/627538c3-fe2c-4d6e-8586-c05678c89e4e), so these test cases are actually wrong.

Reply:
In native type yes, in phpdoc no, as in logical operators, AND has higher priority. This is also "natural".

Reply:
The support should be rather shown like [here](https://psalm.dev/r/bcddbed6bc). OK, maybe it should be parsed in this form too, just like you said - in order to be able to fix it.